examples/client_timeout_test.c

```diff
@@ -0,0 +1,438 @@
+#include <stdlib.h>
```

Comment:
In the long term, the answer is probably yes. I did it this way to avoid modifying the existing client, which by itself has no requirements wrt. to timing. I agree that the timeout could be an argument to the existing client instead of a separate program.